crates/key-server/src/key_server_options.rs

```diff
@@ -1,26 +1,66 @@
 // Copyright (c), Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::errors::InternalError;
 use crate::from_mins;
 use crate::types::Network;
+use anyhow::{anyhow, Result};
 use duration_str::deserialize_duration;
 use semver::VersionReq;
 use serde::{Deserialize, Serialize};
 use std::time::Duration;
 use sui_types::base_types::ObjectID;
+use tracing::info;
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct KeyServerOptions {
-    // TODO: remove this when the legacy key server is no longer needed
-    /// The object ID of the legacy key server object.
-    pub legacy_key_server_object_id: ObjectID,
+/// ClientKeyType for a permissioned client.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub enum ClientKeyType {
+    Derived {
+        derivation_index: u16, // Must be unique
+    },
+    Imported {
+        env_var: String, // Expected a BLS key
```

Comment:
I think this can be typed as a BLS scalar and then tag it to be deserialized from base64. The same thing is done in the CLI.

Reply:
note that this one is the env variable name